src/Accounts/Accounts/Utilities/CommandNotFoundHelper.cs

```diff
@@ -0,0 +1,298 @@
+                }
+            }
+            args.StopSearch = false;
+        }
+
+        private static bool IsAzOrAzureRMCmdlet(string commandName)
+        {
+            return AzOrAzureRMRegex.IsMatch(commandName);
+        }
+
+        private static bool IsAzureRMCommand(string commandName)
+        {
+            return AzureRMRegex.IsMatch(commandName);
+        }
+
+        private static bool TryGetMigrationGuide(string commandName, out MemoryMigrationDetails details)
+        {
+            return LazyCommandToMigrationMappings.Value.TryGetValue(commandName, out details);
+        }
+
+        private static string FormatCommandDeprecationMessage(string commandName, MemoryMigrationDetails details)
+        {
+            string message;
+            if (string.IsNullOrEmpty(details.Replacement))
+            {
+                message = string.Format(Resources.CommandNotFoundDeprecated, commandName, details.Release);
+            }
+            else
+            {
+                message = string.Format(Resources.CommandNotFoundReplaced, commandName, details.Replacement, details.Release);
+            }
+            if (TryGetExternalLinkToMigrationGuide(details.Release, out string link))
+            {
+
+                message += " " + string.Format(Resources.SeeMigrationGuide, link);
+            }
+            return message;
+        }
+
+        private static bool TryGetExternalLinkToMigrationGuide(string release, out string link)
+        {
+            link = null;
+            if (!string.IsNullOrEmpty(release))
+            {
+                // if release is like "Az x.x.x", we can provide a link to migration guide
+                var match = new Regex(@"^Az\s*(\d+\.\d+\.\d+)$", RegexOptions.IgnoreCase).Match(release);
+                if (match.Success && match.Groups.Count >= 2)
+                {
+                    var version = match.Groups[1];
+                    link = $"[todo: fwlink]https://aka.ms/migration?release={version}";
+                }
+            }
+            return !string.IsNullOrEmpty(link);
+        }
+
+        private static bool TryGetModuleOfCommand(string commandName, out string moduleName)
+        {
+            return LazyCommandToModuleMappings.Value.TryGetValue(commandName, out moduleName);
+        }
+
+        private static bool TryGetFuzzyStringSuggestions(string commandName, out IEnumerable<string> suggestions)
+        {
+            var suggestionsInLowercase = FuzzySharp.Process.ExtractTop(
+                commandName.ToLowerInvariant(),
+                LazyCommandLowercaseToNormal.Value.Keys,
+                processor: s => s, // ExtractTop by default converts strings to lowercase.
+                                   // It's redundant work so we override the behavior and cache lowercase command names.
+                scorer: ScorerCache.Get<DefaultRatioScorer>(),
+                cutoff: 90);
```

Comment:
How about using constant to replace this magic number.

Reply:
It only appears once, and I added the optional parameter name ("cutoff: ") to make it more readable, so I think it's OK.